test/computational_graph.jl

```diff
@@ -85,6 +85,39 @@ end
     @test vertices(g2) == external_with_ghost(g2) == external(g2) == [𝑓⁻(1), 𝑏⁻(1), 𝜙(1), 𝑓⁻(2), 𝑓(1), 𝑓⁺(3), 𝜙(1), 𝑏⁺(1), 𝑓⁺(1)]
 end
 
+@testset "Graph Operations" begin
+    V2 = [𝑏⁺(1), 𝑏⁺(2)𝑏⁻(2), 𝑏⁻(3)]
+    g1 =  feynman_diagram(V2, [[1, 2], [3, 4]]; external=[1, 4])
+    g2 = feynman_diagram(V2, [[1, 3], [2, 4]]; external=[1, 4])
+    # print("$(g2.subgraphs), $(g2.operator) \n")
+    g3 = feynman_diagram(V2, [[1, 4], [2, 3]]; external=[1, 4])
+    # print("$(g3.subgraphs), $(g3.operator) \n")
+    gsum = g2 + g3
+    groot =  g1 + gsum
+    # for node in PreOrderDFS(groot)
+    #     for (i, child) in enumerate(children(node))
+    #         print("$(child)\n")
+    #     end
+    # end
+    # print("inplace replace\n")
+    ComputationalGraphs.replace_subgraph!(groot,g2,g3)
+    #ComputationalGraphs.replace_subgraph!(g2.subgraphs[1],g2,g3)
+    gnew = ComputationalGraphs.replace_subgraph(groot,g2,g3)
+    # for node in PreOrderDFS(groot)
+    #     for (i, child) in enumerate(children(node))
+    #         print("$(child)\n")
+    #     end
+    # end
+    # print("copy replace\n")
+    # for node in PreOrderDFS(gnew)
+    #     for (i, child) in enumerate(children(node))
+    #         print("$(child)\n")
+    #     end
+    # end
+
+    @test  isequiv(gsum.subgraphs[1], gsum.subgraphs[2])
+    @test  isequiv(gsum.subgraphs[1], gsum.subgraphs[2] ,:id)
```

Comment:
I suggest to keep only of these two tests, to clearly indicate which behavior is expected (i.e., whether the new subgraph put into groot in place or a copy is generated). Since currently both `replace_subgraph!` and `replace_subgraph` use the subgraph `m` in-place, we should keep line 118.